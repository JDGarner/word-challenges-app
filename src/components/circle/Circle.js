import React from "react";
import { View } from "react-native";
import styled from "styled-components";

const CircleView = styled(View)`
  position: absolute;
  top: ${props => props.top};
  left: ${props => props.left};
  right: ${props => props.right};
  bottom: ${props => props.bottom};
  background-color: ${props => props.color};
  height: ${props => props.radius};
  width: ${props => props.radius};
  border-radius: ${props => props.radius / 2};
`;

const Circle = ({ color, radius, top = 0, left = 0, right = 0, bottom = 0 }) => {
  return (
    <CircleView color={color} radius={radius} top={top} left={left} right={right} bottom={bottom} />
  );
};

export default Circle;
