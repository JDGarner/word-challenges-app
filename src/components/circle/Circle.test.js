import React from "react";
import renderer from "react-test-renderer";
import Circle from "./Circle";

test("renders correctly", () => {
  const tree = renderer.create(<Circle radius={10} color="red" />).toJSON();
  expect(tree).toMatchSnapshot();
});
