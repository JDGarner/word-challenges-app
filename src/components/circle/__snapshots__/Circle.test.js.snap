// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  bottom={0}
  className="sc-bdVaJa ipKhlm"
  color="red"
  left={0}
  radius={10}
  right={0}
  top={0}
/>
`;
